tests/framework/integration.go

```diff
@@ -380,3 +405,23 @@ func getOps(ss []string) ([]clientv3.Op, error) {
 	}
 	return ops, nil
 }
+
+type integrationAuthClient struct {
```

Comment:
Please don't do anything that doesn't make sense in standard client. If `ClusterClientWithAuth` needs to return erorr, then `ClientWithAuth` should also return erorr.

Reply:
> Please don't do anything that doesn't make sense in standard client.

Some auth tests checks unauthenticated errors. 

Integration test will throw an error at client creation stage with `clientv3.New` https://github.com/etcd-io/etcd/blob/7ec336f80c9463fecd6d8ef15a9f05f3e8f6287a/client/v3/client.go#L439

However, e2e test won't fail at client creation stage and fail at "got response" stage. That's why I defer the error from client creation to "got response"stage in integration test. 

Also every request from e2e test will establish a new grpc connection to the server but integration test reuses existing one. In order to use a different user name and password, integration test needs to create another client.

Reply:
What you describe makes sense, however I would prefer different solution to avoid over engineering. It's true that `clientv3.New` used in integration tests will return error if auth is invalid. This is because it calls `Auth.Authenticate` https://github.com/etcd-io/etcd/blob/7ec336f80c9463fecd6d8ef15a9f05f3e8f6287a/client/v3/client.go#L259-L276 when creating client.

Could e2e tests also call `Auth.Authenticate` when creating client with username/password provided?

Reply:
```
dev-dsk-chaochn-2c-a26acd76 % etcdctl auth
NAME:
	auth - Enable or disable authentication

USAGE:
	etcdctl auth <enable or disable> [flags]

API VERSION:
	3.6


COMMANDS:
	disable	Disables authentication
	enable	Enables authentication
	status	Returns authentication status

OPTIONS:
  -h, --help[=false]	help for auth

GLOBAL OPTIONS:
      --cacert=""				verify certificates of TLS-enabled secure servers using this CA bundle
      --cert=""					identify secure client using this TLS certificate file
      --command-timeout=5s			timeout for short running command (excluding dial timeout)
      --debug[=false]				enable client-side debug logging
      --dial-timeout=2s				dial timeout for client connections
  -d, --discovery-srv=""			domain name to query for SRV records describing cluster endpoints
      --discovery-srv-name=""			service name to query when using DNS discovery
      --endpoints=[127.0.0.1:2379]		gRPC endpoints
      --hex[=false]				print byte strings as hex encoded strings
      --insecure-discovery[=true]		accept insecure SRV records describing cluster endpoints
      --insecure-skip-tls-verify[=false]	skip server certificate verification (CAUTION: this option should be enabled only for testing purposes)
      --insecure-transport[=true]		disable transport security for client connections
      --keepalive-time=2s			keepalive time for client connections
      --keepalive-timeout=6s			keepalive timeout for client connections
      --key=""					identify secure client using this TLS key file
      --password=""				password for authentication (if this option is used, --user option shouldn't include password)
      --user=""					username[:password] for authentication (prompt if password is not supplied)
  -w, --write-out="simple"			set the output format (fields, json, protobuf, simple, table)
```

Reply:
I think we can create `clientv3.Client` just for this.